### PR TITLE
feat: Add default views for Course Details and Introduce Your Course …

### PR DESCRIPTION
**Pull request by ruzniaievdm**
### YT: https://youtrack.raccoongang.com/issue/2U-105

### CODE REVIEW

File: src/schedule-and-details/introducing-section/index.jsx [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+        ),
+      }}
+    />
+  );
+
+  return (
+    <section className="section-container details-section">
+      {aboutPageEditable && (
+        <header className="section-header">
+          <span className="lead">
```
Comment (PKulkoRaccoonGang) [inline src/schedule-and-details/introducing-section/index.jsx:71]: Same question for the title. Should the title have an appropriate tag?

File: src/schedule-and-details/introducing-section/index.jsx [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+            {intl.formatMessage(messages.introducingTitle)}
+          </span>
+          <span className="x-small text-gray-700">
```
Comment (PKulkoRaccoonGang) [inline src/schedule-and-details/introducing-section/index.jsx:74]: Same question for description. Should there be an appropriate paragraph tag here?

File: src/schedule-and-details/introducing-section/introduction-video/IntroductionVideo.scss [hunk 3]
```diff
@@ -0,0 +1,11 @@
+.embed-video-container {
+  @include pgn-box-shadow(1, "down");
+
+  box-sizing: border-box;
```
Comment (PKulkoRaccoonGang) [inline src/schedule-and-details/introducing-section/introduction-video/IntroductionVideo.scss:4]: It seems to me that we have a global declaration of this property for all child elements on the page? Please make sure that this property is really needed.

File: src/schedule-and-details/introducing-section/course-card-image/index.jsx [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+const CourseCardImage = ({
+  intl,
+  courseId,
+  courseImageAssetPath,
+  onChange,
+}) => {
+  const dispatch = useDispatch();
+  const uploadedAssets = useSelector(getUploadedAssets);
+  const imageAbsolutePath = assetPathRoot(`/${_.last(courseImageAssetPath.split('/'))}`);
```
Comment (sendr) [inline src/schedule-and-details/introducing-section/course-card-image/index.jsx:34]: why don't you use js util URL, which does all the things with handling extra slashes?
for example
```
const url = new URL(courseImageAssetPath, getConfig().STUDIO_BASE_URL)
url.href
```

Comment (ruzniaievdm) [reply src/schedule-and-details/introducing-section/course-card-image/index.jsx]: It's a good idea, thx

File: src/schedule-and-details/introducing-section/course-card-image/index.jsx [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+
+  const handleChangeImageAsset = (path) => {
+    const assetPath = _.last(path.split('/'));
+    const imageName = _.last(assetPath.split('block@'));
```
[the remark this answers is not in the document]
Comment (ruzniaievdm) [reply src/schedule-and-details/introducing-section/course-card-image/index.jsx:39]: It isn't working with parsing studio urls

File: src/schedule-and-details/introducing-section/utils.js [hunk 6]
```diff
@@ -0,0 +1,6 @@
+import { getConfig } from '@edx/frontend-platform';
+
+const assetPathRoot = (imageAssetPath) => `${getConfig().LMS_BASE_URL}${imageAssetPath}`;
+const assetsUrl = (courseId) => `${getConfig().STUDIO_BASE_URL}/assets/${courseId}`;
```
[the remark this answers is not in the document]
Comment (ruzniaievdm) [reply src/schedule-and-details/introducing-section/utils.js]: Removed file

File: src/schedule-and-details/data/slice.js [hunk 7]
```diff
@@ -11,6 +11,7 @@ const slice = createSlice({
     savingStatus: '',
     courseDetails: {},
     courseSettings: {},
+    uploadAssets: {},
```
[the remark this answers is not in the document]
Comment (ruzniaievdm) [reply src/schedule-and-details/data/slice.js]: Updated

File: src/schedule-and-details/details-section/index.jsx [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+}) => {
+  const formattedLanguage = () => {
+    const result = languageOptions.find((arr) => arr[0] === language);
+    return result ? result[1] : intl.formatMessage(messages.dropdownEmpty);
+  };
+
+  return (
+    <section className="section-container details-section">
+      <header className="section-header">
+        <span className="lead">
```
[the remark this answers is not in the document]
Comment (ruzniaievdm) [reply src/schedule-and-details/details-section/index.jsx:19]: Yeah, you are right it is little bit confusing
But I used this heading in my previous prs
Let's leave this as it is, I will check it in my next pr and update it

File: src/schedule-and-details/details-section/index.jsx [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+          {intl.formatMessage(messages.detailsTitle)}
+        </span>
+        <span className="x-small text-gray-700">
```
[the remark this answers is not in the document]
Comment (ruzniaievdm) [reply src/schedule-and-details/details-section/index.jsx:22]: Answered

File: src/schedule-and-details/details-section/index.jsx [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+import { Form, Dropdown } from '@edx/paragon';
+
+import messages from './messages';
+
+const DetailsSection = ({
+  intl, language, languageOptions, onChange,
+}) => {
+  const formattedLanguage = () => {
+    const result = languageOptions.find((arr) => arr[0] === language);
+    return result ? result[1] : intl.formatMessage(messages.dropdownEmpty);
```
[the remark this answers is not in the document]
Comment (ruzniaievdm) [reply src/schedule-and-details/details-section/index.jsx:13]: But we already have readable function name, I think it is enough

File: src/schedule-and-details/introducing-section/introduction-video/IntroductionVideo.scss (hunk 3 above)
Comment (ruzniaievdm) [reply src/schedule-and-details/introducing-section/introduction-video/IntroductionVideo.scss]: Removed

File: src/schedule-and-details/introducing-section/course-card-image/index.jsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (sendr) [reply src/schedule-and-details/introducing-section/course-card-image/index.jsx:39]: I've just tried the following
```
console.log('-------------- ', _.last(assetPath.split('block@')), ' -> ', _.last(path.split('block@')))
->
--------------  student_to_student.png  ->  student_to_student.png

[the remark this answers is not in the document]
Comment (ruzniaievdm) [reply src/schedule-and-details/introducing-section/course-card-image/index.jsx:39]: we have two options paste input image URL
`http://localhost:18000/asset-v1:edX+M12+2T2023+type@asset+block@а.png`
`/static/а.png`

```
assetPath = asset-v1:edX+M12+2T2023+type@asset+block@image.png
imageName = image.png
```

in your option it looks different
`imageName = static/а.png`

in the former option it works well, but the latter it is not really what I want.

[the remark this answers is not in the document]
Comment (sendr) [reply src/schedule-and-details/introducing-section/course-card-image/index.jsx:39]: ok, let's then add a comment about why we do the separate parsing for '@block'?

**Review by sendr — APPROVED** [review]

**Review by PKulkoRaccoonGang — APPROVED** [review]

**Pull request merged**